htdocs/lib2/logic/user.class.php

```diff
@@ -1605,7 +1605,7 @@ public function foundsUntilNextRating()
         global $opt;
         $rating = $opt['logic']['rating'];
 
-        return (int) ($rating['percentageOfFounds'] - ($this->getStatFound() % $rating['percentageOfFounds']));
+        return ceil($rating['percentageOfFounds'] - ($this->getStatFound() % $rating['percentageOfFounds']));
```

Comment:
Correct me if I'm wrong but why should we use ceil here?
As an example. I have 18 Founds.
10 - (18 % 10)
10 - (8)
2 
So i need 2 more founds for an fav rating.

Reply:
I was thinking of non-integer values of `$rating['percentageOfFounds']` - that's the only case I can imagine here which can produce a result with floating point precision problems.

But just noticed that % with convert non-integer parametes to int.

Then I suggest to either
* completely discard the (int) here, because this calculations cannot produce precision errors, or
* use round() instead.

(int) would produce wrong values for down-jitters, e.g (int) 3.9999999 => 3.